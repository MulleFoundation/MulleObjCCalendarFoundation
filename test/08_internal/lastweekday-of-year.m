#ifdef __MULLE_OBJC__
# import <MulleObjCCalendarFoundation/MulleObjCCalendarFoundation.h>
#else
# import <Foundation/Foundation.h>
#endif


char   *day_name( NSInteger index)
{
   switch( index)
   {
   case 1 : return( "So");
   case 2 : return( "Mo");
   case 3 : return( "Di");
   case 4 : return( "Mi");
   case 5 : return( "Do");
   case 6 : return( "Fr");
   case 7 : return( "Sa");
   }
   return( "???");
}


int   main( void)
{
   NSCalendar                         *calendar;
   NSDateComponents                   *components;
   NSDate                             *date;
   NSInteger                          day;
   NSInteger                          year;
   struct MulleExtendedTimeInterval   extended;

   calendar = [NSCalendar calendarWithIdentifier:NSGregorianCalendar];
   [calendar setTimeZone:[NSTimeZone timeZoneWithAbbreviation:@"GMT"]];
   [calendar setFirstWeekday:2];
   [calendar setMinimumDaysInFirstWeek:4];

   for( year = 1600; year <= 2020; year++)
   {
      components = [[NSDateComponents new] autorelease];
      [components setYear:year];

      date       = [calendar dateFromComponents:components];
      components = [calendar components:~0
                                  fromDate:date];
      MulleExtendedTimeIntervalInit( &extended, [date timeIntervalSinceReferenceDate]);

      printf( "%ld.%ld.%ld ",
            (long) [components day], (long) [components month], (long) [components year]);

      printf( " lastWeekday: %s\n",
               day_name( [calendar mulleLastWeekdayOfYearFromExtendedTimeInterval:&extended]));
   }
   return( 0);
}