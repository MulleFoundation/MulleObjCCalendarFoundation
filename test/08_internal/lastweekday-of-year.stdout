1.1.1600  lastWeekday: So
1.1.1601  lastWeekday: Mo
1.1.1602  lastWeekday: Di
1.1.1603  lastWeekday: Mi
1.1.1604  lastWeekday: Fr
1.1.1605  lastWeekday: Sa
1.1.1606  lastWeekday: So
1.1.1607  lastWeekday: Mo
1.1.1608  lastWeekday: Mi
1.1.1609  lastWeekday: Do
1.1.1610  lastWeekday: Fr
1.1.1611  lastWeekday: Sa
1.1.1612  lastWeekday: Mo
1.1.1613  lastWeekday: Di
1.1.1614  lastWeekday: Mi
1.1.1615  lastWeekday: Do
1.1.1616  lastWeekday: Sa
1.1.1617  lastWeekday: So
1.1.1618  lastWeekday: Mo
1.1.1619  lastWeekday: Di
1.1.1620  lastWeekday: Do
1.1.1621  lastWeekday: Fr
1.1.1622  lastWeekday: Sa
1.1.1623  lastWeekday: So
1.1.1624  lastWeekday: Di
1.1.1625  lastWeekday: Mi
1.1.1626  lastWeekday: Do
1.1.1627  lastWeekday: Fr
1.1.1628  lastWeekday: So
1.1.1629  lastWeekday: Mo
1.1.1630  lastWeekday: Di
1.1.1631  lastWeekday: Mi
1.1.1632  lastWeekday: Fr
1.1.1633  lastWeekday: Sa
1.1.1634  lastWeekday: So
1.1.1635  lastWeekday: Mo
1.1.1636  lastWeekday: Mi
1.1.1637  lastWeekday: Do
1.1.1638  lastWeekday: Fr
1.1.1639  lastWeekday: Sa
1.1.1640  lastWeekday: Mo
1.1.1641  lastWeekday: Di
1.1.1642  lastWeekday: Mi
1.1.1643  lastWeekday: Do
1.1.1644  lastWeekday: Sa
1.1.1645  lastWeekday: So
1.1.1646  lastWeekday: Mo
1.1.1647  lastWeekday: Di
1.1.1648  lastWeekday: Do
1.1.1649  lastWeekday: Fr
1.1.1650  lastWeekday: Sa
1.1.1651  lastWeekday: So
1.1.1652  lastWeekday: Di
1.1.1653  lastWeekday: Mi
1.1.1654  lastWeekday: Do
1.1.1655  lastWeekday: Fr
1.1.1656  lastWeekday: So
1.1.1657  lastWeekday: Mo
1.1.1658  lastWeekday: Di
1.1.1659  lastWeekday: Mi
1.1.1660  lastWeekday: Fr
1.1.1661  lastWeekday: Sa
1.1.1662  lastWeekday: So
1.1.1663  lastWeekday: Mo
1.1.1664  lastWeekday: Mi
1.1.1665  lastWeekday: Do
1.1.1666  lastWeekday: Fr
1.1.1667  lastWeekday: Sa
1.1.1668  lastWeekday: Mo
1.1.1669  lastWeekday: Di
1.1.1670  lastWeekday: Mi
1.1.1671  lastWeekday: Do
1.1.1672  lastWeekday: Sa
1.1.1673  lastWeekday: So
1.1.1674  lastWeekday: Mo
1.1.1675  lastWeekday: Di
1.1.1676  lastWeekday: Do
1.1.1677  lastWeekday: Fr
1.1.1678  lastWeekday: Sa
1.1.1679  lastWeekday: So
1.1.1680  lastWeekday: Di
1.1.1681  lastWeekday: Mi
1.1.1682  lastWeekday: Do
1.1.1683  lastWeekday: Fr
1.1.1684  lastWeekday: So
1.1.1685  lastWeekday: Mo
1.1.1686  lastWeekday: Di
1.1.1687  lastWeekday: Mi
1.1.1688  lastWeekday: Fr
1.1.1689  lastWeekday: Sa
1.1.1690  lastWeekday: So
1.1.1691  lastWeekday: Mo
1.1.1692  lastWeekday: Mi
1.1.1693  lastWeekday: Do
1.1.1694  lastWeekday: Fr
1.1.1695  lastWeekday: Sa
1.1.1696  lastWeekday: Mo
1.1.1697  lastWeekday: Di
1.1.1698  lastWeekday: Mi
1.1.1699  lastWeekday: Do
1.1.1700  lastWeekday: Fr
1.1.1701  lastWeekday: Sa
1.1.1702  lastWeekday: So
1.1.1703  lastWeekday: Mo
1.1.1704  lastWeekday: Mi
1.1.1705  lastWeekday: Do
1.1.1706  lastWeekday: Fr
1.1.1707  lastWeekday: Sa
1.1.1708  lastWeekday: Mo
1.1.1709  lastWeekday: Di
1.1.1710  lastWeekday: Mi
1.1.1711  lastWeekday: Do
1.1.1712  lastWeekday: Sa
1.1.1713  lastWeekday: So
1.1.1714  lastWeekday: Mo
1.1.1715  lastWeekday: Di
1.1.1716  lastWeekday: Do
1.1.1717  lastWeekday: Fr
1.1.1718  lastWeekday: Sa
1.1.1719  lastWeekday: So
1.1.1720  lastWeekday: Di
1.1.1721  lastWeekday: Mi
1.1.1722  lastWeekday: Do
1.1.1723  lastWeekday: Fr
1.1.1724  lastWeekday: So
1.1.1725  lastWeekday: Mo
1.1.1726  lastWeekday: Di
1.1.1727  lastWeekday: Mi
1.1.1728  lastWeekday: Fr
1.1.1729  lastWeekday: Sa
1.1.1730  lastWeekday: So
1.1.1731  lastWeekday: Mo
1.1.1732  lastWeekday: Mi
1.1.1733  lastWeekday: Do
1.1.1734  lastWeekday: Fr
1.1.1735  lastWeekday: Sa
1.1.1736  lastWeekday: Mo
1.1.1737  lastWeekday: Di
1.1.1738  lastWeekday: Mi
1.1.1739  lastWeekday: Do
1.1.1740  lastWeekday: Sa
1.1.1741  lastWeekday: So
1.1.1742  lastWeekday: Mo
1.1.1743  lastWeekday: Di
1.1.1744  lastWeekday: Do
1.1.1745  lastWeekday: Fr
1.1.1746  lastWeekday: Sa
1.1.1747  lastWeekday: So
1.1.1748  lastWeekday: Di
1.1.1749  lastWeekday: Mi
1.1.1750  lastWeekday: Do
1.1.1751  lastWeekday: Fr
1.1.1752  lastWeekday: So
1.1.1753  lastWeekday: Mo
1.1.1754  lastWeekday: Di
1.1.1755  lastWeekday: Mi
1.1.1756  lastWeekday: Fr
1.1.1757  lastWeekday: Sa
1.1.1758  lastWeekday: So
1.1.1759  lastWeekday: Mo
1.1.1760  lastWeekday: Mi
1.1.1761  lastWeekday: Do
1.1.1762  lastWeekday: Fr
1.1.1763  lastWeekday: Sa
1.1.1764  lastWeekday: Mo
1.1.1765  lastWeekday: Di
1.1.1766  lastWeekday: Mi
1.1.1767  lastWeekday: Do
1.1.1768  lastWeekday: Sa
1.1.1769  lastWeekday: So
1.1.1770  lastWeekday: Mo
1.1.1771  lastWeekday: Di
1.1.1772  lastWeekday: Do
1.1.1773  lastWeekday: Fr
1.1.1774  lastWeekday: Sa
1.1.1775  lastWeekday: So
1.1.1776  lastWeekday: Di
1.1.1777  lastWeekday: Mi
1.1.1778  lastWeekday: Do
1.1.1779  lastWeekday: Fr
1.1.1780  lastWeekday: So
1.1.1781  lastWeekday: Mo
1.1.1782  lastWeekday: Di
1.1.1783  lastWeekday: Mi
1.1.1784  lastWeekday: Fr
1.1.1785  lastWeekday: Sa
1.1.1786  lastWeekday: So
1.1.1787  lastWeekday: Mo
1.1.1788  lastWeekday: Mi
1.1.1789  lastWeekday: Do
1.1.1790  lastWeekday: Fr
1.1.1791  lastWeekday: Sa
1.1.1792  lastWeekday: Mo
1.1.1793  lastWeekday: Di
1.1.1794  lastWeekday: Mi
1.1.1795  lastWeekday: Do
1.1.1796  lastWeekday: Sa
1.1.1797  lastWeekday: So
1.1.1798  lastWeekday: Mo
1.1.1799  lastWeekday: Di
1.1.1800  lastWeekday: Mi
1.1.1801  lastWeekday: Do
1.1.1802  lastWeekday: Fr
1.1.1803  lastWeekday: Sa
1.1.1804  lastWeekday: Mo
1.1.1805  lastWeekday: Di
1.1.1806  lastWeekday: Mi
1.1.1807  lastWeekday: Do
1.1.1808  lastWeekday: Sa
1.1.1809  lastWeekday: So
1.1.1810  lastWeekday: Mo
1.1.1811  lastWeekday: Di
1.1.1812  lastWeekday: Do
1.1.1813  lastWeekday: Fr
1.1.1814  lastWeekday: Sa
1.1.1815  lastWeekday: So
1.1.1816  lastWeekday: Di
1.1.1817  lastWeekday: Mi
1.1.1818  lastWeekday: Do
1.1.1819  lastWeekday: Fr
1.1.1820  lastWeekday: So
1.1.1821  lastWeekday: Mo
1.1.1822  lastWeekday: Di
1.1.1823  lastWeekday: Mi
1.1.1824  lastWeekday: Fr
1.1.1825  lastWeekday: Sa
1.1.1826  lastWeekday: So
1.1.1827  lastWeekday: Mo
1.1.1828  lastWeekday: Mi
1.1.1829  lastWeekday: Do
1.1.1830  lastWeekday: Fr
1.1.1831  lastWeekday: Sa
1.1.1832  lastWeekday: Mo
1.1.1833  lastWeekday: Di
1.1.1834  lastWeekday: Mi
1.1.1835  lastWeekday: Do
1.1.1836  lastWeekday: Sa
1.1.1837  lastWeekday: So
1.1.1838  lastWeekday: Mo
1.1.1839  lastWeekday: Di
1.1.1840  lastWeekday: Do
1.1.1841  lastWeekday: Fr
1.1.1842  lastWeekday: Sa
1.1.1843  lastWeekday: So
1.1.1844  lastWeekday: Di
1.1.1845  lastWeekday: Mi
1.1.1846  lastWeekday: Do
1.1.1847  lastWeekday: Fr
1.1.1848  lastWeekday: So
1.1.1849  lastWeekday: Mo
1.1.1850  lastWeekday: Di
1.1.1851  lastWeekday: Mi
1.1.1852  lastWeekday: Fr
1.1.1853  lastWeekday: Sa
1.1.1854  lastWeekday: So
1.1.1855  lastWeekday: Mo
1.1.1856  lastWeekday: Mi
1.1.1857  lastWeekday: Do
1.1.1858  lastWeekday: Fr
1.1.1859  lastWeekday: Sa
1.1.1860  lastWeekday: Mo
1.1.1861  lastWeekday: Di
1.1.1862  lastWeekday: Mi
1.1.1863  lastWeekday: Do
1.1.1864  lastWeekday: Sa
1.1.1865  lastWeekday: So
1.1.1866  lastWeekday: Mo
1.1.1867  lastWeekday: Di
1.1.1868  lastWeekday: Do
1.1.1869  lastWeekday: Fr
1.1.1870  lastWeekday: Sa
1.1.1871  lastWeekday: So
1.1.1872  lastWeekday: Di
1.1.1873  lastWeekday: Mi
1.1.1874  lastWeekday: Do
1.1.1875  lastWeekday: Fr
1.1.1876  lastWeekday: So
1.1.1877  lastWeekday: Mo
1.1.1878  lastWeekday: Di
1.1.1879  lastWeekday: Mi
1.1.1880  lastWeekday: Fr
1.1.1881  lastWeekday: Sa
1.1.1882  lastWeekday: So
1.1.1883  lastWeekday: Mo
1.1.1884  lastWeekday: Mi
1.1.1885  lastWeekday: Do
1.1.1886  lastWeekday: Fr
1.1.1887  lastWeekday: Sa
1.1.1888  lastWeekday: Mo
1.1.1889  lastWeekday: Di
1.1.1890  lastWeekday: Mi
1.1.1891  lastWeekday: Do
1.1.1892  lastWeekday: Sa
1.1.1893  lastWeekday: So
1.1.1894  lastWeekday: Mo
1.1.1895  lastWeekday: Di
1.1.1896  lastWeekday: Do
1.1.1897  lastWeekday: Fr
1.1.1898  lastWeekday: Sa
1.1.1899  lastWeekday: So
1.1.1900  lastWeekday: Mo
1.1.1901  lastWeekday: Di
1.1.1902  lastWeekday: Mi
1.1.1903  lastWeekday: Do
1.1.1904  lastWeekday: Sa
1.1.1905  lastWeekday: So
1.1.1906  lastWeekday: Mo
1.1.1907  lastWeekday: Di
1.1.1908  lastWeekday: Do
1.1.1909  lastWeekday: Fr
1.1.1910  lastWeekday: Sa
1.1.1911  lastWeekday: So
1.1.1912  lastWeekday: Di
1.1.1913  lastWeekday: Mi
1.1.1914  lastWeekday: Do
1.1.1915  lastWeekday: Fr
1.1.1916  lastWeekday: So
1.1.1917  lastWeekday: Mo
1.1.1918  lastWeekday: Di
1.1.1919  lastWeekday: Mi
1.1.1920  lastWeekday: Fr
1.1.1921  lastWeekday: Sa
1.1.1922  lastWeekday: So
1.1.1923  lastWeekday: Mo
1.1.1924  lastWeekday: Mi
1.1.1925  lastWeekday: Do
1.1.1926  lastWeekday: Fr
1.1.1927  lastWeekday: Sa
1.1.1928  lastWeekday: Mo
1.1.1929  lastWeekday: Di
1.1.1930  lastWeekday: Mi
1.1.1931  lastWeekday: Do
1.1.1932  lastWeekday: Sa
1.1.1933  lastWeekday: So
1.1.1934  lastWeekday: Mo
1.1.1935  lastWeekday: Di
1.1.1936  lastWeekday: Do
1.1.1937  lastWeekday: Fr
1.1.1938  lastWeekday: Sa
1.1.1939  lastWeekday: So
1.1.1940  lastWeekday: Di
1.1.1941  lastWeekday: Mi
1.1.1942  lastWeekday: Do
1.1.1943  lastWeekday: Fr
1.1.1944  lastWeekday: So
1.1.1945  lastWeekday: Mo
1.1.1946  lastWeekday: Di
1.1.1947  lastWeekday: Mi
1.1.1948  lastWeekday: Fr
1.1.1949  lastWeekday: Sa
1.1.1950  lastWeekday: So
1.1.1951  lastWeekday: Mo
1.1.1952  lastWeekday: Mi
1.1.1953  lastWeekday: Do
1.1.1954  lastWeekday: Fr
1.1.1955  lastWeekday: Sa
1.1.1956  lastWeekday: Mo
1.1.1957  lastWeekday: Di
1.1.1958  lastWeekday: Mi
1.1.1959  lastWeekday: Do
1.1.1960  lastWeekday: Sa
1.1.1961  lastWeekday: So
1.1.1962  lastWeekday: Mo
1.1.1963  lastWeekday: Di
1.1.1964  lastWeekday: Do
1.1.1965  lastWeekday: Fr
1.1.1966  lastWeekday: Sa
1.1.1967  lastWeekday: So
1.1.1968  lastWeekday: Di
1.1.1969  lastWeekday: Mi
1.1.1970  lastWeekday: Do
1.1.1971  lastWeekday: Fr
1.1.1972  lastWeekday: So
1.1.1973  lastWeekday: Mo
1.1.1974  lastWeekday: Di
1.1.1975  lastWeekday: Mi
1.1.1976  lastWeekday: Fr
1.1.1977  lastWeekday: Sa
1.1.1978  lastWeekday: So
1.1.1979  lastWeekday: Mo
1.1.1980  lastWeekday: Mi
1.1.1981  lastWeekday: Do
1.1.1982  lastWeekday: Fr
1.1.1983  lastWeekday: Sa
1.1.1984  lastWeekday: Mo
1.1.1985  lastWeekday: Di
1.1.1986  lastWeekday: Mi
1.1.1987  lastWeekday: Do
1.1.1988  lastWeekday: Sa
1.1.1989  lastWeekday: So
1.1.1990  lastWeekday: Mo
1.1.1991  lastWeekday: Di
1.1.1992  lastWeekday: Do
1.1.1993  lastWeekday: Fr
1.1.1994  lastWeekday: Sa
1.1.1995  lastWeekday: So
1.1.1996  lastWeekday: Di
1.1.1997  lastWeekday: Mi
1.1.1998  lastWeekday: Do
1.1.1999  lastWeekday: Fr
1.1.2000  lastWeekday: So
1.1.2001  lastWeekday: Mo
1.1.2002  lastWeekday: Di
1.1.2003  lastWeekday: Mi
1.1.2004  lastWeekday: Fr
1.1.2005  lastWeekday: Sa
1.1.2006  lastWeekday: So
1.1.2007  lastWeekday: Mo
1.1.2008  lastWeekday: Mi
1.1.2009  lastWeekday: Do
1.1.2010  lastWeekday: Fr
1.1.2011  lastWeekday: Sa
1.1.2012  lastWeekday: Mo
1.1.2013  lastWeekday: Di
1.1.2014  lastWeekday: Mi
1.1.2015  lastWeekday: Do
1.1.2016  lastWeekday: Sa
1.1.2017  lastWeekday: So
1.1.2018  lastWeekday: Mo
1.1.2019  lastWeekday: Di
1.1.2020  lastWeekday: Do
